app/src/main/java/org/openobservatory/ooniprobe/common/PreferenceManager.java

```diff
@@ -98,10 +100,25 @@ public boolean isManualUploadDialog() {
 		return sp.getBoolean(IS_MANUAL_UPLOAD_DIALOG, true);
 	}
 
+	/*
+	 * This method is used to ask user to enable push notifications.
+	 */
+	public boolean isAskNotificationDialog() {
+		return sp.getBoolean(IS_NOTIFICATION_DIALOG, true);
+	}
+
 	public boolean isNotifications() {
 		return sp.getBoolean(r.getString(R.string.notifications_enabled), false);
 	}
 
+	public void setNotificationsFromDialog(boolean notifications) {
+		//set notification value and increment app open
+		sp.edit()
+				.putLong(IS_NOTIFICATION_DIALOG, getAppOpenCount()+1)
```

Comment:
Why are we incrementing the count here? It does not seem necessary when setting the notifications?

Reply:
We do the same thing in iOS, we increment it to don't do it next time the app is opened.

Reply:
There are many isomorphic way to implement it, I guess.